src/components/CharacterAssignment.jsx

```diff
@@ -0,0 +1,233 @@
+        newColumns.unassignedCharacters.lineCount += reassignedCounts;
+        const index = newColumnList.findIndex(id => id == key);
+        newColumnList.splice(index, 1);
+        delete newColumns[key];
+      }
+    }
+    setColumns(newColumns);
+    setColumnList(newColumnList);
+  };
+
+  /**
+   * Creates the initial actor columns and the unassigned column when a script hasn't been selected yet.
+   * NOTE: When a script isn't selected, these columns are hidden. It is still important to set them up in state
+   *  so they can be populated when we do have a current script.
+   * @param actors - Array of objects with properties of name and id
+   */
+  const createInitialActorColumns = (actors) => {
+    const actorColumns = {
+      'unassignedCharacters': {
+        id: 'unassignedCharacters',
+        title: 'Unassigned Characters',
+        characterIds: [],
+        lineCount: 0
+      }
+    };
+    const actorIds = [];
+    actors.forEach((actor) => {
+      actorColumns[actor.id] = {
+        id: actor.id.toString(),
+        title: actor.name,
+        characterIds: [],
+        lineCount: 0
+      };
+      actorIds.push(actor.id.toString());
+    });
+    setColumns(actorColumns);
+    setColumnList(actorIds);
+  };
+
+  /**
+   * Removes a character from its source column and adds it to its destination column,
+   *  updates the database with the assignment info, and sets state to reflect the changes.
+   * @param result - React-beautiful-dnd set parameter
+   */
+  const onDragEnd = async (result) => {
+    const { destination, source, draggableId } = result;
+    // No destination, change nothing
+    if (!destination) {
+      return;
+    }
+    // Dropped back in same location, change nothing
+    if (destination.droppableId === source.droppableId) {
+      return;
+    }
+    // Get the column ids of the source and destination columns
+    const start = columns[source.droppableId];
+    const finish = columns[destination.droppableId];
+
+    // Remove the character from the column it was picked up from and update the column values
+    const index = start.characterIds.indexOf(draggableId);
+    start.characterIds.splice(index, 1);
+    start.lineCount = start.characterIds.reduce((acc, curr) => {
+      return acc + characters[curr].lineCount;
+    }, 0);
+
+    // Add the character to the column it was dropped on
+    finish.characterIds.push(draggableId);
+    finish.lineCount = finish.characterIds.reduce((acc, curr) => {
+      return acc + characters[curr].lineCount;
+    }, 0);
+
+    // Update the assigned character in the database and if it works, reset the column values to represent the assignment
+    const response = await assignCharacter(draggableId, destination.droppableId, currentScriptId);
+    if (response.success) {
+      setColumns({ ...columns, [start.id]: start, [finish.id]: finish });
```

Comment:
The bracket notation allows an expression as the key name instead of a string.
Here's more about it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors